kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -202,14 +204,13 @@ private synchronized void handleChunkAssignment(CacheSlotMetadata cacheSlotMetad
       }
 
       registerSearchMetadata(snapshotMetadata.name);
-      successfulChunkAssignments.increment();
+      assignmentTimer.stop(chunkAssignmentTimerSuccess);
```

Comment:
timers should always be closed to capture the elapsed time. Otherwise it results in weird metrics data. Further, counting success and failures from timers is an odd pattern in metrics.

I think it's better to leave the counters as is and add a timer to capture the time for chunk assignment and chunk evictions.

Reply:
For timer samples, there's no need (nor option) to close them - [per the documentation](https://micrometer.io/docs/concepts#_storing_start_state_in_timer_sample) the expectation is that after stopping a timer sample it will go out of scope.

If we want I can keep the existing counters, but the [micrometer documentation](https://micrometer.io/docs/concepts#_counters) specifically calls this out as an anti-pattern and recommends against using the combination of timers and counters:

> Never count something you can time with a Timer or summarize with a DistributionSummary! Both Timer and DistributionSummary always publish a count of events in addition to other measurements.
> 
> Be sure to read through the timer section before jumping into using counters, as timers record a count of timed events as part of the suite of metrics that go into timing. For those pieces of code you intend to time, you do NOT need to add a separate counter.

Lastly, encoding the status (ie success, fail) is the approach suggested by micrometer for [various](https://micrometer.io/docs/concepts#_multi_gauge) [types](https://micrometer.io/docs/concepts#_storing_start_state_in_timer_sample) of instrumentation.

Reply:
Didn't realize Timer also emitted counts. Thought it was only the distribution summary. Thanks for the doc link.